@Test public void shouldRejectDuplicateEntryWhenUsingPopulatingUpdater() throws Exception {
  populator=newPopulator();
  addUpdate(populator,1,"value1");
  addUpdate(populator,2,"value2");
  Value value=Values.of("value1");
  when(nodePropertyAccessor.getNodePropertyValue(1,PROPERTY_KEY_ID)).thenReturn(value);
  when(nodePropertyAccessor.getNodePropertyValue(3,PROPERTY_KEY_ID)).thenReturn(value);
  try {
    IndexUpdater updater=populator.newPopulatingUpdater(nodePropertyAccessor);
    updater.process(add(3,schemaDescriptor,"value1"));
    updater.close();
    fail("should have thrown exception");
  }
 catch (  IndexEntryConflictException conflict) {
    assertEquals(1,conflict.getExistingNodeId());
    assertEquals(value,conflict.getSinglePropertyValue());
    assertEquals(3,conflict.getAddedNodeId());
  }
}
